tests/Doctrine/Tests/ORM/Functional/QuotedIdentifierAssociationTest.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace Doctrine\Tests\ORM\Functional;
+
+use Doctrine\Tests\Models\Quote\User;
+use Doctrine\Tests\Models\Quote\UserData;
+
+/**
+ * Tests that association with quoted JoinColumn is loaded
+ */
+class QuotedIdentifierAssociationTest extends \Doctrine\Tests\OrmFunctionalTestCase
+{
+	protected function setUp()
+	{
+		$this->useModelSet('quote');
+		parent::setUp();
+	}
+	public function testUpdateEntityWithIdentifierAssociationWithQuotedJoinColumn()
+	{
+		$user = new User();
+		$user->name = 'John Doe';
+		$this->_em->persist($user);
+		$this->_em->flush();
+
+		$userData = new UserData();
+		$userData->name = '123456789';
+		$userData->user = $user;
+		$this->_em->persist($userData);
+		$this->_em->flush();
+
+		$userData->name = '4321';
+		$this->_em->flush();
+
+		$queries = $this->_sqlLoggerStack->queries;
+		$this->assertSQLEquals('UPDATE `quote-user-data` SET `name` = ? WHERE `user-id` = ?', $queries[$this->_sqlLoggerStack->currentQuery - 1]['sql']);
```

Comment:
Quoting depends on the platform, so the backticks are not valid with all the various phpunit setups we have for different platforms. In postgresql (and other standard RDBMSs), identifiers are quoted via `"`, for example.

The test will need to be adapted as following, in my opinion:

1. quote all these 3 identifiers (`quote-user-data`, `name` and `user-id`)
2. expect them to NOT be equals to the non-quoted version
3. assert on the actual performed query with the quoted identifiers passed in via `sprintf()`